### DISCUSSION TITLE
Persist and Non-Persistent Stores Together

### DISCUSSION THREAD

**Discussion by willcray** [pmndrs/zustand#593] [Q&A]
I'm trying to make an app that utilizes both a persistent and non-persistent store. As a workaround, I'm currently making both of my stores persistent:

```
const options = {
    name: 'persistent-storage',
    getStorage: () => AsyncStorage
}

const useStore = create(persist( (set: any, get: any) => ({
    ...createStore(set, get),
    ...createPersistentStore(set, get),
}),
    options
))
```

But I would like to make it so only the store defined in createPersistentStore is persistent. Is there a way to define createStore as non-persistent and createPersistentStore as persistent, while being able access them both through the useStore hook?

Thanks!

**Comment by dai-shi** [comment]
Use yet-to-be-documented `pertialize` option?
https://github.com/pmndrs/zustand/blob/d9780c78e54f08f699a36b60727968c2c3895f0f/src/middleware.ts#L234-L239

**Reply by willcray** [reply]
I can't seem to apply `partialize` to the non-persistent `createStore()` call. This gives me an error since I'm not passing `set` to `createStore()` in `partialize: createStore()`, but I've tried other syntax that results in none of the stores being persistent. Right now, if I comment out the partialize line of `persistOptions`, then it works, but the state created by createStore is persistent, and I don't want it to be. Additional documentation around partialize would be helpful.

```
const createStore = (set: any, get: any): Store => ({
    ...initialState,
    resetState: () => {
        set(initialState)
    },
    setImage: (newImage: IMAGE) => {
        set({
            image: newImage
        })
    },
   ...
})

// TODO: make the "createStore" half of the store not persistent
const persistOptions = {
    name: 'persistent-storage',
    getStorage: () => AsyncStorage,
    onRehydrateStorage: () => onRehydrate(),
    partialize: () => createStore()
    
}

const useStore = create(persist((set: any, get: any) => ({
    ...createStore(set, get),
    ...createPersistentStore(set, get),
}),
    persistOptions
))
```

**Reply by dai-shi** [reply]
`partialize` should be a pure function. Something like this?
```js
partialize: (state) => Object.fromEntries(Object.keys(initialState).map((key) => [key, state[key]]))
```
